### PR TITLE
Transform JSX components to TSX

### PR DESCRIPTION
**Pull request by ghost**
#### Pre-flight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/develop/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://github.com/ferdium/ferdium-app/issues) for a feature request that matches the one I want to file, without success.

#### Description of Change

- chore: update vscode settings
- fix: SetupAssistant types& lint issues
- chore: turn WorkspaceDrawer to tsx
- refactor: ImportantScreen types& format
- chore: turn SettingsLayout to tsx
- chore: turn ServiceWebView to tsx
- chore: turn Services & appear to tsx
- chore: turn SetupAssistant to tsx
- chore: turn Import to tsx
- refactor: color picker types & file structure

#### Motivation and Context

- [ Convert to typescript #42 ](https://github.com/ferdium/ferdium-app/issues/42)

#### Checklist

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`npm run prepare-code`)
- [x] `npm test` passes
- [x] I tested/previewed my changes locally

### CODE REVIEW

File: src/containers/auth/ImportScreen.tsx [hunk 1]
```diff
@@ -3,7 +3,12 @@ import { inject, observer } from 'mobx-react';
 import { StoresProps } from '../../@types/ferdium-components.types';
 import Import from '../../components/auth/Import';
 
-class ImportScreen extends Component<StoresProps> {
+interface IProps extends StoresProps {}
+
+//  TODO - [TECH DEBT][NOT TESTED][PROPER UI NAVIGATION NOT FOUND] Need to verify if this page is still working.
```
Comment (vraravam) [inline src/containers/auth/ImportScreen.tsx]: when you are logged out of Ferdium, there will be a screen where you can import settings.

Comment (ghost) [reply src/containers/auth/ImportScreen.tsx]: ok will test this screen and update here

Comment (ghost) [reply src/containers/auth/ImportScreen.tsx]: is it some automatic logout?
I have tried manual logout after feridium account login, it redirect me to initial screen i.e

image

Comment (vraravam) [reply src/containers/auth/ImportScreen.tsx]: after logging out, go to "use without server", and then there should be something to import.

Comment (ghost) [reply src/containers/auth/ImportScreen.tsx]: is there a way to clear app data ?
as I have tried both login and use without server they seems to be loading separate sessions
intermediate screen for import is not loading

Comment (vraravam) [reply src/containers/auth/ImportScreen.tsx]: other than wiping out your whole profile folder, i dont know any other mechanism

Comment (ghost) [reply src/containers/auth/ImportScreen.tsx]: how to do that in debug mode let me try it there first?

**Comment by vraravam** [comment]
`rm -rf ~/Library/Application\ Support/FerdiumDev/`

File: src/containers/auth/ImportScreen.tsx (hunk 1 above)
Comment (vraravam) [reply src/containers/auth/ImportScreen.tsx]: `rm -rf ~/Library/Application\ Support/FerdiumDev/`

Comment (ghost) [reply src/containers/auth/ImportScreen.tsx]: Marking PR as draft till I confirm the screen.
It didn't worked out. Still seeing start screen same as above after login and logout even after using the command.
am Im missing something ?
tried app with or without rm -rf ~/Library/Application\ Support/FerdiumDev/ and clean build(packaged) , debug

Comment (vraravam) [reply src/containers/auth/ImportScreen.tsx]: ok - im out of ideas. If someone can figure out where this screen comes, we can hold this PR. If not, lets time-box the hold, and merge this in - say in 48 hrs?

Comment (ghost) [reply src/containers/auth/ImportScreen.tsx]: ok sure

Comment (ghost) [reply src/containers/auth/ImportScreen.tsx]: As 48 hours over , I've rebased the pr and marked ready for merge

Comment (Alphrag) [reply src/containers/auth/ImportScreen.tsx]: I did try to get to that page which, according to `route.tsx`, is located at `/auth/signup/import` by tricking one of the redirection from the welcome page, but it didn't show me anything and I haven't seen it anywhere either in accountless mode, or by creating an account (could it be only for custom servers?).

Comment (vraravam) [reply src/containers/auth/ImportScreen.tsx]: If its dead code, please go ahead and remove it in this PR itself.

Comment (ghost) [reply src/containers/auth/ImportScreen.tsx]: ok i shall remove !

Comment (ghost) [reply src/containers/auth/ImportScreen.tsx]: @vraravam removed import and its component!

File: .vscode/extensions.json [hunk 2]
```diff
@@ -1,8 +1,31 @@
 {
     "recommendations": [
-      "esbenp.prettier-vscode",
-      "dbaeumer.vscode-eslint",
-      "codezombiech.gitignore",
-      "EditorConfig.EditorConfig"
+        "esbenp.prettier-vscode",
```
Comment (vraravam) [inline .vscode/extensions.json]: According to the `.editorconfig` file already present in the repo, indentation should be at 2 spaces - not 4. Please change

File: src/components/services/content/ServiceWebview.tsx [hunk 3]
```diff
@@ -114,7 +122,7 @@ class ServiceWebview extends Component {
             });
           }, 0);
         }}
-        onUpdateTargetUrl={this.updateTargetUrl}
+        // onUpdateTargetUrl={this.updateTargetUrl} // TODO - [TS DEBT] need to check where its from
```
Comment (vraravam) [inline src/components/services/content/ServiceWebview.tsx:125]: will this affect the "open link in Ferdium" capability? Or if the user edits the service recipe, and changes some url setting - will that reload be affected by commenting this out?

**Review by vraravam — CHANGES_REQUESTED** [review]

File: .vscode/extensions.json (hunk 2 above)
Comment (ghost) [reply .vscode/extensions.json]: fixed it pushing in other commit.
[suggestion] editor config is not enforced like eslint in future may we can configure as part of eslint

File: src/components/services/content/ServiceWebview.tsx (hunk 3 above)
Comment (ghost) [reply src/components/services/content/ServiceWebview.tsx:125]: No, This property is not existing in current class as well as it is not passed or assigned.
when from js to ts conversion I found it stale and can't find its purpose but I marked it with comment and leaving variable for now incase I found its purpose way down further refactoring i can come back and update 

 will check the lib what is actually props intended to perform

https://www.electronjs.org/docs/latest/api/webview-tag#event-update-target-url 
the is, actual behavior of prop listener (onUpdateTargetUrl)

File: .vscode/extensions.json (hunk 2 above)
Comment (ghost) [reply .vscode/extensions.json]: amend to last commit and pushed!

File: src/components/services/content/ServiceWebview.tsx (hunk 3 above)
Comment (muhamedsalih-tw) [reply src/components/services/content/ServiceWebview.tsx:125]: will check the lib what is actually props intended to perform

Comment (muhamedsalih-tw) [reply src/components/services/content/ServiceWebview.tsx:125]: https://www.electronjs.org/docs/latest/api/webview-tag#event-update-target-url 

this is actual behavoiur of prop listener (onUpdateTargetUrl) but may due to some bulk change the function has been lost

**Review by vraravam — APPROVED** [review]

**Pull request merged**